manifests/yum/repo/local_os_updates.pp

```diff
@@ -1,77 +1,95 @@
 #
 # * For more complex scenarios, create a site-specific profile and use the native
 #   `yumrepo` type directly.
 #
+#  @example Describing a single server with specific URLs
+#    # This explicitly sets the `baseurl` and `gpgkey` keys in os_updates.repo.
+#    # (This overrides all other parameters and automagic URL logic.)
+#    simp::yum::repo::os_updates_local {
+#      baseurl => 'https://yum.test.simp/yum/CentOS/8/x86_64/Updates',
+#      gpgkey  => 'https://yum.test.simp/yum/SIMP/GPGKEYS/RPM-GPG-KEY-CentOS-8',
+#    }
+#
 #  @example Describing a single server by FQDN
-#    # When classified to an CentOS 6 x86_64 host, this creates an os_updates
-#    # yumrepo with the ``baseurl`` "https://yum.test.simp/yum/CentOS/6/x86_64/Updates"
+#    # When classified to an CentOS 7 x86_64 host, this creates an os_updates
+#    # yumrepo with the `baseurl` "https://yum.test.simp/yum/CentOS/7/x86_64/Updates"
 #    simp::yum::repo::os_updates_local {
 #      servers => ['yum.test.simp']
 #    }
 #
 #  @example Describing a single server by FQDN
-#    # When classified to an CentOS 6 x86_64 host, this creates an os_updates
-#    # yumrepo with a 3-entry ``baseurl`` and a 3-entry ``gpgkey``
+#    # When classified to an CentOS 7 x86_64 host, this creates an os_updates
+#    # yumrepo with a 3-entry `baseurl` and a 3-entry `gpgkey`
 #    simp::yum::repo::os_updates_local {
 #      servers => [
 #        'yum.test.simp',
 #        'yum2.test.simp',
-#        'https://yum.updates.url/full/path/to/repo/c6-64-u'
+#        'https://yum.updates.url/specific/path/to/repo/c7-64-u'
 #      ],
-#      extra_gpgkey_urls => [
-#        'https://yum.updates.url/full/path/to/repo/c6-64-u/RPM-GPG-KEY-CentOS-6'
-#      ]
+#      gpgkey => 'https://yum.updates.url/full/path/to/repo/c6-64-u/RPM-GPG-KEY-CentOS-7',
 #    }
 #
 # @param servers
 #   An Array of FQDNs, IPs, or URLs containing the yum server(s) to use.
 #
 #   * An FQDN or IP will be assumed to host it yum repository and GPG keys at
-#     the URLs established by ``simp::server::yum``.
+#     the URLs established by `simp::server::yum`.
 #
 #   * A URL will be used as-is, and should point directly to its yum repository.
 #
+#   This parameter has no effect if the `baseurl` parameter is set directly.
+#
 # @param enable_repo
 #   Enables or disables the Yum repo
 #
 # @param extra_gpgkey_urls
-#   An optional Array of Urls to include additional GPG key files
+#   An optional Array of Urls to include additional GPG key files.
+#   This parameter has no effect if the `gpgkey` parameter is set directly.
+#
+# @param relative_repo_path
+#   The relative path to the yum repo relative to the URL(s) set in `$servers`.
+#   This parameter has no effect if the `baseurl` parameter is set directly.
+#
+# @param baseurl
+#   The URL for this repository. Set this to absent to remove it from the file completely.
+#   Set this parameter directly to completely skip all automated URL logic.
 #
+# @param gpgkey
```

Comment:
why is this parameter needed when extra_gpgkey_urls could be used?

Reply:
@lnemsick-simp Would you prefer something like `disable_defaults` or the like? That would make `extra_gpgkey_urls` the only thing that works. Looking back at this, it's sort of confusing across the board but we needed some way to make things "just work".

Reply:
@trevor-vaughan How about default_gpgkey_urls as an Array[Simplib::URI]

Reply:
After re-reading the original, as well as all of this, I think that it makes sense to push the new `gpgkey` parameter back into the code.

It looks like using `servers` (with URLs as confusing as that is) combined with `extra_gpgkeys` should give people everything that they need.

Changing `servers` to `default_gpgkeys` probably makes sense but is API breaking so we can figure out if we want to flip it with a compat option or just leave it for now.